### ISSUE TITLE
tabs the callback of select function

### ISSUE THREAD

**Issue by watonyweng** [angular-ui/bootstrap#5777]
The issues forum is __NOT__ for support requests. It is for bugs and feature requests only.
Please read https://github.com/angular-ui/bootstrap/blob/master/CONTRIBUTING.md and search
existing issues (both open and closed) prior to opening any new issue and ensure you follow the instructions therein.

### Version of Angular, UIBS, and Bootstrap

Angular: 1.4.4

UIBS: 1.3.1

Bootstrap: 3.3.6

**Comment by wesleycho** [comment]
Please post a reproduction via Plunker.

**Comment by 1203460788** [comment]
@wesleycho ; this plunker link http://plnkr.co/edit/TUlFYebCnfs8kqxlVd0P?p=preview;

**Comment by watonyweng** [comment]
@wesleycho  ; this plunker is [http://plnkr.co/edit/TUlFYebCnfs8kqxlVd0P?p=preview;](http://plnkr.co/edit/TUlFYebCnfs8kqxlVd0P?p=preview;)

**Comment by icfantv** [comment]
@1203460788, @watonyweng, please exercise patience.  this is an OSS project and we are all unpaid volunteers.  we will get to the issue when we are able to make time.  duplicate posts and pings are considered rude.  thanks.

**Comment by wesleycho** [comment]
@icfantv your commit hash is broken there.

**Comment by wesleycho** [comment]
I'm confused with the Plunker - that is exactly what the select callback is for. This doesn't sound like a bug.

Please explain in more detail as per GitHub issue template.

**Comment by RobJacobs** [comment]
I think the issue is he feels the select callback should not get fired if the tab is already active/selected.

**Comment by Hsensor** [comment]
@RobJacobs ! you are right. may be my english is poor.

**Comment by icfantv** [comment]
We've already had a request to ensure it DOES fire even if already selected so this would be a breaking change.  I'm nearly certain we pass the `$event` object so one can very easily check to see if the current tab is already selected and call `preventDefault()`.

I don't have a strong preference as to what solution we go with, but it's far easier to cancel the event than to trigger the select again...

**Comment by wesleycho** [comment]
The $event object is only present if it is triggered by an event - if it is a model change, then one would not have access to the event object.

**Comment by Hsensor** [comment]
if I use /\bactive\b/.test($event.$event.target.parentNode.ClassName)==true in the on select callback  ,I only need to return prevent other code run ,not is calling preventDefault . so I must do it onDeselect callback like select callback,because I found the source code ctrl.select will call onSelect and Deselect,but the previousIndex==oldIndex at the moment .
`ctrl.select = function(index, evt) {
    if (!destroyed) {
      var previousIndex = findTabIndex(oldIndex);
      var previousSelected = ctrl.tabs[previousIndex];
      if (previousSelected) {
        previousSelected.tab.onDeselect({
          $event: evt,
          $selectedIndex: index
        });
        if (evt && evt.isDefaultPrevented()) {
          return;
        }
        previousSelected.tab.active = false;
      }

      var selected = ctrl.tabs[index];
      if (selected) {
        selected.tab.onSelect({
          $event: evt
        });
        selected.tab.active = true;
        ctrl.active = selected.index;
        oldIndex = selected.index;
      } else if (!selected && angular.isDefined(oldIndex)) {
        ctrl.active = null;
        oldIndex = null;
      }
    }
  };` 

I can't understand is that why angular-bootstrap team give the way to allow the selected tab can be fired again.

**Comment by icfantv** [comment]
@wesleycho, that's true.  But if one is triggering it manually via code, then I think we have to assume the dev knows what they are doing.  Either way, I don't think that is what is being asked for here - though it is something to consider.

**Comment by icfantv** [comment]
@Hsensor, can you sum up in one or two sentences what exactly the behavior is you expect vs. the exhibited behavior?  Thanks.

**Comment by RobJacobs** [comment]
It's easy enough to have a reference to the active tab in the controller then check that against the selected tab index to determine whether you need to take action or not (that's how I handle this scenario).  Here is a [plunk](http://plnkr.co/edit/0cfJWoVHO7OkhAvGsl9p?p=preview) that demonstrates that approach.

**Comment by yoshiokatsuneo** [comment]
@RobJacobs I cannot open the link.

**Comment by sh977218** [comment]
+1,  the select function got called every time you select, even it's already selected.
